docs/howto/manage-web-app-charms/use-a-database.rst

```diff
@@ -0,0 +1,335 @@
+.. _use-db-with-12-factor-charms:
+
+Use a database with your 12-factor app charm
+============================================
+
+This guide demonstrates how to integrate a MySQL database into your FastAPI
+12-factor app. You will learn how to define object-relational mappings (ORM),
+handle schema migrations with Alembic, and manage the database integration using Juju.
+
+Prerequisites
+-------------
+
+- A working FastAPI application charmed with 12-factor tooling. This guide will be
+  following the :ref:`tutorial up until you deploy the app
+  <deploy-the-fastapi-app>`.
+- A Juju model
+- Basic familiarity with `SQLAlchemy <https://www.sqlalchemy.org/>`_ and
+  `Alembic <https://alembic.sqlalchemy.org/en/latest/>`_.
+
+
+Prepare the environment
+-----------------------
+
+MySQL requires a specific Python driver. Ensure your ``requirements.txt`` includes
+a MySQL driver compatible with SQLAlchemy, such as ``pymysql``.
+
+.. code-block::
+    :caption: requirements.txt
+
+    fastapi
+    sqlalchemy
+    alembic
+    PyMySQL[rsa]
+
+.. warning::
+
+  When using ``pymysql``, your connection string (handled later by the charm)
+  generally follows the format ``mysql+pymysql://user:pass@host/db``.
```

Comment:
I don't feel that this merits a warning.

If we can make it so pymysql is the connector, then I'd make this part of the main text:


```suggestion
When using pymysql, your connection string (handled later by the charm)
follows the format ``mysql+pymysql://user:pass@host/db``.
```